xray/utils/resultstable.go

```diff
@@ -478,3 +483,175 @@ func getOperationalRiskViolationReadableData(violation services.Violation) *oper
 		newerVersions: newerVersions,
 	}
 }
+
+// simplifyVulnerabilities returns a new slice of services.Vulnerability that contains only the unique vulnerabilities from the input slice
+// The uniqueness of the vulnerabilities is determined by the getUniqueKey function
+func simplifyVulnerabilities(scanVulnerabilities []services.Vulnerability, multipleRoots bool) []services.Vulnerability {
+	var uniqueVulnerabilities = make(map[string]*services.Vulnerability)
+	for _, vulnerability := range scanVulnerabilities {
+		for vulnerableComponentId := range vulnerability.Components {
+			vulnerableDependency, vulnerableVersion, _ := SplitComponentId(vulnerableComponentId)
+			packageKey := getUniqueKey(vulnerableDependency, vulnerableVersion, vulnerability.Cves, len(vulnerability.Components[vulnerableComponentId].FixedVersions) > 0)
+			if uniqueVulnerability, exist := uniqueVulnerabilities[packageKey]; exist {
+				fixedVersions := appendUniqueFixVersions(uniqueVulnerability.Components[vulnerableComponentId].FixedVersions, vulnerability.Components[vulnerableComponentId].FixedVersions...)
+				impactPaths := appendUniqueImpactPaths(uniqueVulnerability.Components[vulnerableComponentId].ImpactPaths, vulnerability.Components[vulnerableComponentId].ImpactPaths, multipleRoots)
+				uniqueVulnerabilities[packageKey].Components[vulnerableComponentId] = services.Component{
+					FixedVersions: fixedVersions,
+					ImpactPaths:   impactPaths,
+				}
+				continue
+			}
+			uniqueVulnerabilities[packageKey] = &services.Vulnerability{
+				Cves:       vulnerability.Cves,
+				Severity:   vulnerability.Severity,
+				Components: map[string]services.Component{vulnerableComponentId: vulnerability.Components[vulnerableComponentId]},
+				IssueId:    vulnerability.IssueId,
+				Technology: vulnerability.Technology,
+			}
+		}
+	}
+	// convert map to slice
+	result := make([]services.Vulnerability, 0, len(uniqueVulnerabilities))
+	for _, v := range uniqueVulnerabilities {
+		result = append(result, *v)
+	}
+	return result
+}
+
+func SimplifyViolations(scanViolations []services.Violation, multipleRoots bool) []services.Violation {
```

Comment:
Let's add a description for this function.

Reply:
done